src/telliot_core/gas/legacy_gas.py

```diff
@@ -86,13 +95,34 @@ async def legacy_gas_station(chain_id: int, speed: Optional[ethgastypes] = None,
             logger.error(f"Error fetching gas price: {e}")
             return None
 
-    if speed not in prices:
-        logger.error(f"Invalid gas price speed for gasstation: {speed}")
-        return None
-
-    if prices[speed] is None:
-        logger.error("Unable to fetch gas price from gasstation")
-        return None
+    if isinstance(speed, tuple):
```

Comment:
I think we can refactor this logic of retrieving the price like in #358 , as an example